POC/ColorPicker/ColorPicker/MainWindow.xaml.cs

```diff
@@ -47,21 +54,35 @@ private void OnTransparentScreenClick(object sender, EventArgs e)
             DeactivateColorSelectionMode();
         }
 
-        private void OnColorButtonClick(object sender, EventArgs e)
+        private void OnTransparentScreenEscape(object sender, EventArgs e)
+        {
+            SetColor(_previousColor);
+            DeactivateColorSelectionMode();
+        }
+
+        private void OnKeyDown(object sender, KeyEventArgs e)
         {
-            if (_isColorSelectionEnabled)
+            if (e.Key == Key.Escape)
             {
-                DeactivateColorSelectionMode();
+                OnTransparentScreenEscape(sender, e);
             }
-            else
+        }
+
+        private void OnNewColorButtonClick(object sender, EventArgs e)
+        {
+            if (NewColorButton.IsChecked ?? false)
             {
                 ActivateColorSelectionMode();
             }
+            else
+            {
+                DeactivateColorSelectionMode();
+            }
         }
 
         private void UpdateCurrentColor(object sender, EventArgs e)
         {
-            if (_isColorSelectionEnabled)
+            if (NewColorButton.IsChecked ?? false)
```

Comment:
Just my opinion, but this syntax is less common and has not materially shorter than NewColorButton.isChecked != null && NewColorButton. Prefer keeping it simple where possible.

Reply:
Resolving because this was discussed verbally among ourselves (the null-coalescing operator is commonplace in C#).